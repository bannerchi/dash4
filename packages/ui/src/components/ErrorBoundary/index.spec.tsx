/// <reference types="@types/jest" />
import React, { useState } from 'react';
import { render } from 'react-testing-library';
import { ErrorBoundary } from './index';

function BuggyCounter({ initialCounter = 0 }: { initialCounter: any }) {
	const [counter, setCounter] = useState(initialCounter);

	function handleClick() {
		setCounter(counter + 1);
	}

	if (typeof counter !== 'number') {
		// Simulate a JS error
		throw new Error('counter have to be a number type');
	}

	return <h1 onClick={handleClick}>{counter}</h1>;
}

describe('ErrorBoundary', () => {
	test('exist', () => {
		expect(typeof ErrorBoundary).toBe('object');
	});
	test('render default', () => {
		// tslint:disable-next-line
		const consoleError = console.error;
		// tslint:disable-next-line
		console.error = () => {};
		const { container } = render(
			<ErrorBoundary>
				<BuggyCounter initialCounter="0" />
			</ErrorBoundary>
		);
		expect(container.firstChild).toThrowErrorMatchingSnapshot();
		// tslint:disable-next-line
		console.error = consoleError;
	});
	test('render error with custom title', () => {
		// tslint:disable-next-line
		const consoleError = console.error;
		// tslint:disable-next-line
		console.error = () => {};
		const { container } = render(
			<ErrorBoundary title="cunstom title">
				<BuggyCounter initialCounter="0" />
			</ErrorBoundary>
		);
		expect(container.firstChild).toThrowErrorMatchingSnapshot();
		// tslint:disable-next-line
		console.error = consoleError;
	});
	test('render error with custom message', () => {
		// tslint:disable-next-line
		const consoleError = console.error;
		// tslint:disable-next-line
		console.error = () => {};
		const { container } = render(
			<ErrorBoundary message="cunstom message">
				<BuggyCounter initialCounter="0" />
			</ErrorBoundary>
		);
		expect(container.firstChild).toThrowErrorMatchingSnapshot();
		// tslint:disable-next-line
		console.error = consoleError;
	});
});
