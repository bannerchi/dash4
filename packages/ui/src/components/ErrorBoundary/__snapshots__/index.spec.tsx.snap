// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary render default 1`] = `"received is not a function"`;

exports[`ErrorBoundary render error with custom message 1`] = `"received is not a function"`;

exports[`ErrorBoundary render error with custom title 1`] = `"received is not a function"`;
